lib/bson/objectid.js

```diff
@@ -12,13 +12,6 @@ var MACHINE_ID = parseInt(Math.random() * 0xffffff, 10);
 // Regular expression that checks for hex value
 var checkForHexRegExp = new RegExp('^[0-9a-fA-F]{24}$');
 
-// Check if buffer exists
```

Comment:
I find all of this a bit confusing (not because of your work, but the original code).  So we have this check to ensure that we know if `Buffer` exists or not, but then later we use a `var _Buffer = Buffer;` and now we're switching to use `if (Buffer == null)` checks.

Is there a way we can clean this up a bit more?  Also, have you guaranteed that the replacement for the `Buffer.from` (using hex constructor) is always available in node 4+?

Reply:
so it looks like `new Buffer(string, encoding)` exists Node 4+, but is deprecated starting with Node 6. meanwhile `Buffer.from(string, encoding)` exists [Node 4.5+](https://nodejs.org/docs/latest-v4.x/api/buffer.html#buffer_class_method_buffer_from_str_encoding). so I guess we should try to use` Buffer.from`, and only fall back on `new Buffer` if `Buffer.from` doesn't exist?